### ISSUE TITLE
UI - mockups for sharing problem url and adding new problem

### ISSUE THREAD

**Issue by afijal** [benetech/MathShare#364]
Create mockups to show proposed design of sharing problem url and creating new problem

**Comment by afijal** [comment]
Share button in top-right corner
![image](https://user-images.githubusercontent.com/2682379/44038206-8dc0c344-9f16-11e8-9e70-ebb026034bf5.png)
Share popup modal
![image](https://user-images.githubusercontent.com/2682379/44038236-a4c45510-9f16-11e8-8d6d-0d8cad226467.png)

I guess that it would be good to display that dialog after user presses done as well

**Comment by afijal** [comment]
See problem #8 
![image](https://user-images.githubusercontent.com/2682379/44038331-d98f9336-9f16-11e8-9f50-c90cf60774ab.png)

After pressing #8
![image](https://user-images.githubusercontent.com/2682379/44038529-483a3df4-9f17-11e8-8ff4-2b78f8291f0a.png)

After entering new problem initial steps:
![image](https://user-images.githubusercontent.com/2682379/44038500-37267726-9f17-11e8-9ab9-131b640a1e89.png)

And then after pressing Done we could either ask user whether he is sure to add new problem to problem set or save it instantly

**Comment by oskarhinc** [comment]
@afijal We need to add a possibility to add problem title and text value as well

**Comment by johnhbenetech** [comment]
@oskarhinc @afijal Yes, I think it would essentially entail adding the existing symbol palette and an input to the header bar. Visually it might make sense to put it behind an 'edit' button at the type so it's not showing the entire time.

This is related to #105, but I think this can proceed assuming that work will be completed in the future

**Comment by oskarhinc** [comment]
A user should be able to remove unwanted problems from problem set, so maybe adding an "X" button on each of them would be a good solution. 
![screenshot from 2018-08-14 09-52-01](https://user-images.githubusercontent.com/19305104/44078852-db3a3fba-9fa7-11e8-8559-8f8db5872427.png)
 There also should be a "share" button visible, and it would be nice if the "new problem" button would have some kind of plus sign here. Of course, the colors and the layout are just mock, so it needs adjustments, but the result might be similar to the provided screenshot.

**Comment by afijal** [comment]
@johnhbenetech 
New landing page (add step is always last tile)
![image](https://user-images.githubusercontent.com/2682379/44103506-b6a7be6c-9fec-11e8-86fc-ece3df71bb49.png)


And a few questions:
-should delete and add problem buttons look like now, or do we want them more colourfull?
-when creating new problem, can teacher add multiple steps as a 'starting point'? It matters a lot when taking adding multiple problems at once into consideration
-we want to remove done button and replace it with save button. I wonder whether there is a need for share button and save button to exist together. What would be a difference between them? As to share url we need to save current revision anyway, so share button would actually save as well.
So maybe we would like to have both edit and share url in header (initial value: "not saved" or whatever) and update them upon saving, and leave "done" button somewhere?
What are your thoughts on that?

**Comment by afijal** [comment]
@johnhbenetech 
I've updated my question, if you could answer by Thursday it would be great (we have holliady here tomorrow)

**Comment by johnhbenetech** [comment]
@afijal @oskarhinc with regards to adding and deleting - what do we think of something like the following?

![problemset](https://user-images.githubusercontent.com/12927553/44120778-4d1ab2fc-9fd2-11e8-9659-daa82b767262.png)

**Comment by johnhbenetech** [comment]
@afijal 
-when creating new problem, can teacher add multiple steps as a 'starting point'? It matters a lot when taking adding multiple problems at once into consideration

No. I think we can assume a teacher will create:
*  a single problem set
** containing multiple problems
*** which each contain a single prompt

**Comment by johnhbenetech** [comment]
-we want to remove done button and replace it with save button. I wonder whether there is a need for share button and save button to exist together. What would be a difference between them? 

I think saving and sharing should be different.

Sharing is a specific action in which the user intends to have another person look at the record. After this action they expect to be afforded something that they would send to that person. I think almost every application we use today has the 'share' concept, and in almost every case this causes a link to pop up. I think we should try to emulate this in our share action.


Saving is more complicated. Purely from a user aspect, I think the ideal interaction is auto-save:

* There is always an edit URL associated with your record. Any action you take is automatically associated with that URL. You can leave at any time and when you return to that URL you pick up where you left off.

We had discussed that the autosave may put undue stress on the system (making dozens of DB calls every minute for each user). Additionally it may use up a lot of URLs that were really just people testing the system (with no intention of returning to that problem).


So - I think for now we need to keep them as separate actions. We can start with having the save action tied to a button, and revisit if we feel we want to move to something more autosave-y.
